### ISSUE TITLE
Allow create arrows with curved lines.

### ISSUE THREAD

**Issue by fredygamez** [flameshot-org/flameshot#2057]
Allow create arrows with curved lines.

**Comment by veracioux** [comment]
@fredygamez You mean something like this?

![image](https://user-images.githubusercontent.com/29044423/141340401-5371bcd5-8035-4b11-903c-9ecac1a336a3.png)

**Comment by fredygamez** [comment]
Yes. I do.

**Comment by fredygamez** [comment]
![image](https://user-images.githubusercontent.com/17584642/141387379-e0c8e0d9-7480-4d2e-869b-cc02f0712ed7.png)

**Comment by mmahmoudian** [comment]
I think one way to implement this is to have a handle in the middle of the arrow. User draw the arrow as usual (a straight one) and then if they want to make it curved, then they select the arrow and drag the handle which is in the middle of the line to create the curve. The benefits of this approach:
1. I think it is relatively easy to implement
2. Doesn't add new options to config
3. Doesn't need us to add additional button to gui
4. The arrow works as usual so it does not break the current behaviour

**Comment by samuelmissonnier** [comment]
Hello, I saw the project and I tried the different features of the app, I found this issue and I would like to contribute to this, I already forked the project and cloneed it to test it with QT CREATOR, I've the technicals skills with C++ and QT Creator, I realized 4 project with included two PoC.

**Comment by borgmanJeremy** [comment]
Yup! If you have a question you can reach out to us: https://matrix.to/#/#Flameshot:feneas.org

**Comment by samuelmissonnier** [comment]
OK, thank you !

**Comment by santiFerra** [comment]
have someone made it? bc I would like to do it.

**Comment by mmahmoudian** [comment]
@santiFerra check out #2129
